java-frontend/src/main/java/org/sonar/java/se/FlowComputation.java

```diff
@@ -323,13 +324,28 @@ public int hashCode() {
         SymbolicValue peekValue = edge.child.programState.peekValue();
         if (peekValue instanceof SymbolicValue.ExceptionalSymbolicValue) {
           Type type = ((SymbolicValue.ExceptionalSymbolicValue) peekValue).exceptionType();
-          String msg = String.format("%s is thrown.", type == null ? "Exception" : ("'" + type.name() + "'"));
+          String msg = String.format("%s is thrown.", exceptionName(type));
           return Optional.of(location(edge.parent, msg));
         }
       }
       return Optional.empty();
     }
 
+    private Optional<JavaFileScannerContext.Location> flowFromCaughtException(ExplodedGraph.Edge edge) {
+      ProgramPoint programPoint = edge.parent.programPoint;
+      if (((CFG.Block) programPoint.block).isCatchBlock() && programPoint.i == 0) {
+        VariableTree catchVariable = ((VariableTree) programPoint.syntaxTree());
+        SymbolicValue.CaughtExceptionSymbolicValue caughtSv = ((SymbolicValue.CaughtExceptionSymbolicValue) edge.child.programState.getValue(catchVariable.symbol()));
+        Type exceptionType = caughtSv.exception().exceptionType();
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') A "NullPointerException" could be thrown; "caughtSv" is nullable here. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2259)

Reply:
I would add a `Precondition` here to enforce that non-null is expected.